test/e2e/shoot/internal/logging/event_logging.go

```diff
@@ -0,0 +1,264 @@
+}
+
+// After cleans all resources created in Prepare function
+func (v *EventLoggingVerifier) After(ctx context.Context) {
+	By("Delete seed and shoot events")
+	Eventually(func(g Gomega) {
+		g.Expect(v.ShootFramework.SeedClient.Client().Delete(ctx, v.seedEvent)).To(Succeed())
+		g.Expect(v.ShootFramework.ShootClient.Client().Delete(ctx, v.shootEvent)).To(Succeed())
+	}).Should(Succeed())
+}
+
+func getEventFor(clusterType, namespace string) *corev1.Event {
+	clusterType = strings.ToLower(clusterType)
+
+	event := &corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind:      clusterType,
+			Namespace: namespace,
+			Name:      clusterType + "-" + eventLoggerName + "-testing",
+		},
+		Reason: eventLoggerName + "-" + clusterType + "-testing",
+		Type:   "Normal",
+		Source: corev1.EventSource{
+			Component: eventLoggerName,
+			Host:      clusterType,
+		},
+		FirstTimestamp: metav1.Time{
+			Time: time.Now(),
+		},
+	}
+
+	switch clusterType {
+	case "seed":
+		event.Name = seedEventName
+		event.Message = eventMessageForSeed
+	case "shoot":
+		event.Name = shootEventName
+		event.Message = eventMessageForShoot
+	default:
+		event.Name = "unknown-event"
+		event.Message = "Event from Event Logger integration test related to unknown cluster type"
+	}
+
+	return event
+}
+
+// waitUntilLokiReceivesEvent waits until the loki instance in <lokiNamespace> receives <wantedEventMessages> for filtered events by <queryAfterUnpack>.
+// If wantedEventMessages is empty than the function will return nill and will not query the Loki for any events.
+func waitUntilLokiReceivesEvent(ctx context.Context, logger logr.Logger, interval time.Duration, lokiLabels map[string]string, tenant, lokiNamespace, queryAfterUnpack string, wantedEventMessages []string, startTime time.Time, c kubernetes.Interface) error {
+	if len(wantedEventMessages) < 1 {
+		return nil
+	}
+
+	query := `{job="event-logging"} | unpack`
+	if queryAfterUnpack != "" {
+		query = query + ` | ` + queryAfterUnpack
+	}
+	log := logger.WithValues("query", query)
+
+	return retry.Until(ctx, interval, func(ctx context.Context) (done bool, err error) {
+		search, err := getLokiLogsWithCMD(ctx, log, lokiLabels, tenant, lokiNamespace, query, startTime, c)
+		if err != nil {
+			return retry.SevereError(err)
+		}
+
+		if len(search.Data.Result) < len(wantedEventMessages) {
+			log.Info("Waiting to receive all expected logs", "events", len(search.Data.Result), "wanted", len(wantedEventMessages))
+			return retry.MinorError(fmt.Errorf("received only %d/%d events", len(search.Data.Result), len(wantedEventMessages)))
+		} else if len(search.Data.Result) > len(wantedEventMessages) {
+			return retry.SevereError(fmt.Errorf("expected to receive %d events but was %d", len(wantedEventMessages), len(search.Data.Result)))
+		}
+
+		var aggErr errorsutil.AggregateError
```

Comment:
I don't think this is the type you wanted to import, it's coming from `onsi/gomega`